Refactored `ButtonsWidget` and `AlignWidget`. New `blockWidth`, `blockAlignment` and `size` widgets based on ButtonsWidget. (#7555)
plone/volto: docs/source/reference/widgets.md

---
myst:
  html_meta:
    "description": "Volto provides a set of widgets that provide structural features."
    "property=og:description": "Volto provides a set of widgets that provide structural features."
    "property=og:title": "Volto widgets"
    "keywords": "Plone, Volto, widgets"
---

# Widgets

> Volto provides a set of widgets that provide structural features.

Reviewer: WTF does "structural features" mean?

## `ButtonsWidget`

> This component is a helper for building widgets that have a list of buttons that can be toggled in order to select a single value.

Reviewer 2: ```suggestion
This component is a helper for building widgets that have a list of buttons that can be toggled similar to a radio input, allowing the selection of a single value only.
```

   */
  value?: ActionValue;

  /**
   * Default value of the widget.
   */
  default?: ActionValue;

  /**
   * Indicates if the widget is disabled.
   */
  disabled?: boolean;

  /**
   * Indicates if the widget is disabled (alternative flag for compatibility reasons).
   */
  isDisabled?: boolean;
};
```

## `blockWidth`

`blockWidth` is a widget to select a width from the defined `config.blocks.widths`.
It's based on the `ButtonsWidget` under the hood, so the actions and the styles to be applied are configurable.

```{image} /_static/blockWidth.png
:alt: blockWidth
```

## `blockAlignment`

`blockAlignment` is a widget to select the block alignment, one of either `left`, `right`, or `center`.
It's based on the `ButtonsWidget` under the hood, so the actions and the styles to be applied are configurable.

```{image} /_static/blockAlignment.png
:alt: BlockAlignment
```

## `size`

`size` is a widget to select the block size from a default list of values, one of either `small`, `medium`, or `large`.
It's based on the `ButtonsWidget` under the hood, so the actions and the styles to be applied are configurable.

```{image} /_static/size.png
:alt: size

> ```

Reviewer 2: `````suggestion
````{card}
```{image} ../_static/size.png
:alt: size
:target: ../_static/size.png
```
+++
_`size`_
````
`````

---
myst:
  html_meta:
    "description": "Volto provides a set of widgets that provide structural features."

>     "property=og:description": "Volto provides a set of widgets that provide structural features."

Reviewer 2: Align with revisions in narrative text

```suggestion
    "description": "Volto widgets"
    "property=og:description": "Volto widgets"
```
